vip-jetpack/vip-jetpack.php

```diff
@@ -716,3 +716,37 @@ function vip_remove_jetpack_search_menu_page() {
 	);
 }
 add_action( 'admin_menu', 'vip_remove_jetpack_search_menu_page', PHP_INT_MAX );
+
+/**
+ * Account for X-Mobile-Class header in jetpack_is_mobile()
+ *
+ * @param bool|string $matches Boolean if current UA matches $kind or not. If
+ * $return_matched_agent is true, should return the UA string
+ * @param string      $kind Category of mobile device being checked. Can be 'any', 'smart' or 'dumb'.
+ * @param bool        $return_matched_agent Boolean indicating if the UA should be returned
+ *
+ * @return bool|string $matches Boolean if current UA matches $kind or not. If
+ * $return_matched_agent is true, should return the UA string
+ */
+function vip_jetpack_is_mobile( $matches, $kind, $return_matched_agent ) {
+	if ( ! isset( $_SERVER['HTTP_X_MOBILE_CLASS'] ) ) {
+		// No value set, return early.
+		return $matches;
+	}
+
+	// phpcs:ignore Squiz.PHP.CommentedOutCode.Found
+	$x_mobile_class = sanitize_text_field( $_SERVER['HTTP_X_MOBILE_CLASS'] ); // "desktop", "smart", "dumb", "tablet"
+
+	if ( 'desktop' === $x_mobile_class ) {
+		return false;
+	}
+
+	if ( 'smart' === $kind || 'dumb' === $kind ) {
+		$matches = $kind === $x_mobile_class;
+	} elseif ( 'any' === $kind ) {
+		$matches = true;
```

Comment:
If `$return_matched_agent` is true, shouldn't we return the UA string? At least the doc block says so...

Reply:
Good catch, updated now!